code/processes/gateway.q

```diff
@@ -454,8 +454,19 @@ asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
    if[99h<>type servertype;
      // its a list of servertypes e.g. `rdb`hdb
      servertype:distinct servertype,();
-     missing:servertype except exec distinct servertype from .gw.servers where active;
-     if[count missing; errStr:.gw.errorprefix,"not all of the requested server types are available; missing "," " sv string missing];
+     //list of active servers
+     activeservers:exec distinct servertype from .gw.servers where active;
+     //list of all servers
+     allservers:exec distinct servertype from .gw.servers;
+     //if any requested servers are missing then:
+       //if requested server does not exist, return error with list of available servers
+       //if requested server exists but is currently inactive, return error with list of available servers
+     if[count servertype except activeservers; 
+       $[max not servertype in allservers; 
+	errStr:.gw.errorprefix,"the following are not valid servers: ",(", " sv string servertype except allservers),". Available servers include: "," " sv string activeservers;
```

Comment:
inconsistent indenting on this line which uses tab compared to surrounding which use spaces

Reply:
Will fix that